src/libstore/unix/build/local-derivation-goal.hh

```diff
@@ -307,6 +307,16 @@ struct LocalDerivationGoal : public DerivationGoal
      * rewrites caught everything
      */
     StorePath makeFallbackPath(OutputNameView outputName);
+
+    void chmod_(const Path & path, mode_t mode);
+
+    /**
+     * Move/rename path 'src' to 'dst'. Temporarily make 'src' writable if
+     * it's a directory and we're not root (to be able to update the
+     * directory's parent link "..").
+     */
+    void movePath(const Path & src, const Path & dst);
+    void replaceValidPath(const Path & storePath, const Path & tmpPath);
```

Comment:
Not sure if `movePath` and `chmod_` should be member functions.

Reply:
I would keep it as is. Putting private things in headers is an unfortunate bummer with C++, so I rather not do it except for when necessary.